Unix/tests/cli/test_cli.cpp

```diff
@@ -339,10 +343,45 @@ static int StopServerSudo()
 #else
     if (!skipTest)
     {
-        std::stringstream cmd;
-        cmd << "sudo kill -15 " << (int)serverProcess.reserved;
-        if (system(cmd.str().c_str()) == -1)
+        uint args = 0;
+        const char* argv[MAX_SERVER_ARGS];
+        std::stringstream pidStr;
+        int pid;
+        Process killProcess;
+        int status;
+        int numWaits = 0;
+
+        if (PIDFile_Read(&pid) != 0)
             return -1;
+
+        pidStr << pid;
+
+        argv[args++] = sudoPath;
+        argv[args++] = "-E";
+        argv[args++] = "kill";
+        argv[args++] = "-term";
+        argv[args++] = pidStr.str().c_str();
+        argv[args++] = NULL;
+
+        if (Process_StartChild(&killProcess, sudoPath, (char**)argv) != 0)
+            return -1;
+        
+try_again:
```

Comment:
I'm not wild on `goto` when a `while` loop would have done it. But I know that this is done commonly in OMI ...